samples/kotlin-sample/src/main/java/com/apollographql/apollo/kotlinsample/data/ApolloCoroutinesService.kt

```diff
@@ -67,7 +67,7 @@ class ApolloCoroutinesService(
     job = CoroutineScope(processContext).launch {
       try {
         val response = apolloClient.query(commitsQuery).toDeferred().await()
-        val headCommit = response.data?.viewer?.repository?.ref?.target?.asCommit
+        val headCommit = response.data?.viewer?.repository?.ref?.target as GithubRepositoryCommitsQuery.Commit?
```

Comment:
Bumped into this while merging master. This feels more verbose than the previous solution. Maybe for inline fragments we can keep the `asXyz` helpers?